v2.1.0-rc* clean slate builds fail due to unresolvable commit ca8daa2870f3

### Checklist

- [X] This is **not** a question or a support request. If you have any boost related questions, please ask in the [discussion forum](https://github.com/filecoin-project/boost/discussions).
- [X] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/boost/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [X] This is **not** an enhancement request. If it is, please file a [improvement suggestion](https://github.com/filecoin-project/boost/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fenhancement&template=enhancement.yml) instead.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/boost/issues) and the [discussion forum](https://github.com/filecoin-project/boost/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/boost/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to boost.

### Boost component

- [X] boost daemon - storage providers
- [X] boost client
- [X] boost UI
- [X] boost data-transfer
- [X] boost index-provider
- [X] Other

### Boost Version

```text
v2.1.0-rc3
v2.1.0-rc2
and probably others
```


### Describe the Bug

On the morning of 2023-11-09 building boost, with the steps outlined in the [installation instructions](https://boost.filecoin.io/getting-started), started failing. Troubleshooting the issue pointed to a go module dependency `github.com/filecoin-project/boost/extern/boostd-data@v0.0.0-20231009154452-ca8daa2870f3` that wasn't resolvable any more, neither by go's hosted module cache nor on Github itself. Checking in the repo and on github the reason for this turned out to be that commit `ca8daa2870f3` wasn't referenced by the repo any more.

Thankfully after contacting @masih the commit could be reintroduced to a branch of the repo resolving the issue.

Slack discussion: 
[first](https://filecoinproject.slack.com/archives/C03CKDLEWG1/p1699542807096979) 
[resolution](https://filecoinproject.slack.com/archives/C03CKDLEWG1/p1699611349518479)

### Logging Information

```text
The build errors look like this:

> cmd/boost/wallet_cmd.go:18:2: github.com/filecoin-project/boost/extern/boostd-data@v0.0.0-20231009154452-ca8daa2870f3: invalid version: unknown revision ca8daa2870f3
> cmd/boost/retrieve_cmd.go:24:2: github.com/filecoin-project/boost/extern/boostd-data@v0.0.0-20231009154452-ca8daa2870f3: invalid version: unknown revision ca8daa2870f3
```


### Repo Steps

To isolate the issue I created a Dockerfile to make static, clean slate builds of boost.

NOTE: as the commit in question has been made resolvable again on the `filecoin-project/boost` repo builds are working again so the error described here will not appear. If one where to make a similar situation though, this Dockerfile could be used to illustrate the issue.

```Dockerfile
FROM ubuntu:22.04

# Deps & Build toolchain ######################################################
RUN apt-get update \
    && apt-get install -y mesa-opencl-icd \
                          ocl-icd-opencl-dev \
                          gcc \
                          git \
                          bzr \
                          jq \
                          pkg-config \
                          curl \
                          clang \
                          build-essential \
                          hwloc \
                          libhwloc-dev \
                          wget \
                          ca-certificates \
                          gnupg \
    && rm -rf /var/lib/apt/lists/*

# Node16
# https://github.com/nodesource/distributions#installation-instructions
ENV NODE_MAJOR=16
RUN mkdir -p /etc/apt/keyrings \
    && curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg \
    && echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_$NODE_MAJOR.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list \
    && apt-get update -y \
    && apt-get install -y nodejs \
    && rm -rf /var/lib/apt/lists/*

# Go
ARG BOOST_GO_VERSION=1.21.4
RUN curl -L -o go.tar.gz \
            https://go.dev/dl/go${BOOST_GO_VERSION}.linux-amd64.tar.gz \
         && rm -rf /usr/local/go \
         && tar -C /usr/local -xzf go.tar.gz

ENV PATH=$PATH:/usr/local/go/bin:/root/go/bin
RUN go version

# Rust
ENV CARGO_HOME=/usr/local/cargo
ENV RUSTUP_HOME=/usr/local/rustup
RUN curl -o /usr/local/bin/rustup \
        --proto '=https' \
        --tlsv1.2 \
        -sSf https://sh.rustup.rs \
    && chmod +x /usr/local/bin/rustup \
    && rustup -y

ENV PATH=$PATH:/usr/local/cargo/bin

ENV RUSTFLAGS="-C target-cpu=native -g"
ENV FFI_BUILD_FROM_SOURCE=1

###############################################################################

# Boost
ARG BOOST_BUILD_TARGET
ARG BOOST_VERSION=v2.1.0-rc3
WORKDIR /root
RUN git clone https://github.com/filecoin-project/boost \
    && cd boost \
    && git checkout $BOOST_VERSION \
    && go mod tidy \
    && make clean calibnet \
    && make install

EXPOSE 8044

```

1. Run 'docker build -f Dockerfile.static -t boost-static .'
2. See error `github.com/filecoin-project/boost/extern/boostd-data@v0.0.0-20231009154452-ca8daa2870f3: invalid version: unknown revision ca8daa2870f3`
...

--- a/extern/boostd-data/go.mod
+++ b/extern/boostd-data/go.mod
@@ -58,7 +58,7 @@ require (
 	github.com/go-stack/stack v1.8.1 // indirect
 	github.com/gogo/protobuf v1.3.2 // indirect
 	github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da // indirect
-	github.com/golang/protobuf v1.5.2 // indirect
+	github.com/golang/protobuf v1.5.3 // indirect
 	github.com/golang/snappy v0.0.4 // indirect
 	github.com/gorilla/websocket v1.5.0 // indirect
 	github.com/hailocab/go-hostpool v0.0.0-20160125115350-e80d13ce29ed // indirect

--- a/go.mod
+++ b/go.mod
@@ -320,7 +320,7 @@ require (
 require (
 	github.com/filecoin-project/boost-gfm v1.26.7
 	github.com/filecoin-project/boost-graphsync v0.13.9
-	github.com/filecoin-project/boost/extern/boostd-data v0.0.0-20231009154452-ca8daa2870f3
+	github.com/filecoin-project/boost/extern/boostd-data v0.0.0-20231101173716-1622d0ce2581
 	github.com/filecoin-project/go-data-transfer/v2 v2.0.0-rc7
 	github.com/filecoin-project/go-fil-markets v1.28.3
 	github.com/filecoin-project/lotus v1.25.0-rc1

